src/main/java/mate/academy/bookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.bookstore.repository.impl;
+
+import java.util.List;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.repository.BookRepository;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        try {
+            sessionFactory.inTransaction(session -> session.persist(book));
+            return book;
+        } catch (Exception e) {
+            throw new RuntimeException("Can't create new book", e);
```

Comment:
`RuntimeException` is a bad design - consider creating your own custom exception

and don't forget, that now you're working on a complex project - you should design your exception handling accordingly. 

think of whether you need a generic exception like `DataAccessException` for all the repositories, or you will create separate exceptions like `BookNotFoundException`, `BookNotSavedException`, `OrderNotFoundException` etc. etc.

it is only up to you how to organize the exception handling :)